polaris-react/src/components/OptionList/components/Option/Option.tsx

```diff
@@ -138,7 +138,21 @@ export function Option({
         blockAlign={verticalAlignToBlockAlign(verticalAlign)}
       >
         {mediaMarkup}
-        {label}
+        <span
+          style={{
+            fontWeight: 'var(--p-font-weight-semibold)',
+            minWidth: `${String(label).length ?? undefined}ch`,
```

Comment:
Whoops. Just thinking this through a bit more, I need to check if the type is a string before running this. Since it can be a react node as well, we could end up with this length check counting markup as well as the label and we'd end up with a ridiculous min width here.

i.e.
```ts
const node = <div>node</div>;
console.log(String(node).length, 'node'.length); // 15 4
```